tests/bluetooth/host/buf/bt_buf_get_type/src/main.c

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/zephyr.h>
+#include <zephyr/bluetooth/hci.h>
+#include <zephyr/bluetooth/buf.h>
+#include "kconfig.h"
+#include "buf_help_utils.h"
+
+/* Rows count equals number of types */
+#define TEST_PARAMETERS_LUT_ROWS_COUNT		7
+
+/*
+ *  LUT containing testing parameters that will be used
+ *	during each iteration to cover different scenarios
+ */
+static const enum bt_buf_type testing_params_lut[TEST_PARAMETERS_LUT_ROWS_COUNT] = {
+	/** HCI command */
+	BT_BUF_CMD,
+	/** HCI event */
+	BT_BUF_EVT,
+	/** Outgoing ACL data */
+	BT_BUF_ACL_OUT,
+	/** Incoming ACL data */
+	BT_BUF_ACL_IN,
+	/** Outgoing ISO data */
+	BT_BUF_ISO_OUT,
+	/** Incoming ISO data */
+	BT_BUF_ISO_IN,
+	/** H:4 data */
+	BT_BUF_H4,
+};
```

Comment:
If you want to verify the size of the array, it's not enough to specify its length. You will not get a compiler warning if the specified length is larger than the initializer.

```suggestion
static const enum bt_buf_type testing_params_lut[] = {
	/** HCI command */
	BT_BUF_CMD,
	/** HCI event */
	BT_BUF_EVT,
	/** Outgoing ACL data */
	BT_BUF_ACL_OUT,
	/** Incoming ACL data */
	BT_BUF_ACL_IN,
	/** Outgoing ISO data */
	BT_BUF_ISO_OUT,
	/** Incoming ISO data */
	BT_BUF_ISO_IN,
	/** H:4 data */
	BT_BUF_H4,
};
BUILD_ASSERT(ARRAY_SIZE(testing_params_lut) == TEST_PARAMETERS_LUT_ROWS_COUNT);
```

Reply:
Actually, the array size was specified to be used with 'LISTIFY' asit doesn't accept the 'ARRAY_SIZE' as a parameter.

On the other hand, trying to exceed the specified size will result in 'excess elements in array initializer' warning.

But your suggestion is a good one, I didn't know about it.

Reply:
yep I thought it'd work but the definition of ARRAY_SIZE is C code, and not a C macro, so the preprocessor doesn't actually know the size of the array.

Reply:
(Joking) It can always be solved with [more macros].

[more macros]: https://www.boost.org/doc/libs/1_39_0/libs/preprocessor/doc/ref/list_size.html